### PR TITLE
Add 3 name restriction edge cases

### PR DESCRIPTION
**Pull request by bhtowles**
# Description of change
Add 3 edge test cases for field name, collection name, and namespace length restrictions.

# QA steps
 - [ ] automated tests passing
 - [ ] manual qa steps passing (list below)
 
# Risks

# Rollback steps
 - revert this branch

### CODE REVIEW

File: tests/test_mongodb_fname_restrictions.py [hunk 1]
```diff
@@ -0,0 +1,285 @@
...
+            last_index = client['simple_db']['simple_coll_2'].find().count()
+            client["simple_db"]["simple_coll_2"].insert_one({"int.field": 100, "string$field": random_string_generator()})
+            object_id = client["simple_db"]["simple_coll_2"].find()[last_index]['_id']
+            changed_ids.add(object_id)
+
+            last_index += 1
+            client["simple_db"]["simple_coll_2"].insert_one({"int.field": 101, "string$field": random_string_generator()})
+            object_id = client["simple_db"]["simple_coll_2"].find()[last_index]['_id']
+            changed_ids.add(object_id)
+
```
Comment (kspeer825) [inline tests/test_mongodb_fname_restrictions.py:248]: Why the insert for the `$` fields, and the update for the `.` fields? Is this because of the client limitations we discussed?

File: tests/test_mongodb_fname_restrictions.py [hunk 2]
```diff
@@ -0,0 +1,285 @@
...
+        record_count_by_stream = runner.examine_target_output_file(self,
+                                                                   conn_id,
+                                                                   self.expected_sync_streams(),
+                                                                   self.expected_pks())
+
+        # Verify that we got at least 6 records due to changes
+        # (could be more due to overlap in gte oplog clause)
+        for k,v in record_count_by_stream.items():
+            self.assertGreaterEqual(v, 6)
+
```
Comment (kspeer825) [inline tests/test_mongodb_fname_restrictions.py:278]: I think there should be another assertion to validate the document values in the collections so we ensure the special characters are read parsed correctly by the tap.

**Review by kspeer825 — CHANGES_REQUESTED** [review]

File: tests/test_mongodb_fname_restrictions.py [hunk 3]
```diff
@@ -0,0 +1,285 @@
...
+RECORD_COUNT = {}
+
+
+def random_string_generator(size=6, chars=string.ascii_uppercase + string.digits):
+    return ''.join(random.choice(chars) for x in range(size))
+
+def generate_simple_coll_docs(num_docs):
+    docs = []
+    for int_value in range(num_docs):
+        # BUG TDL-20088. Version 4.2 does not support leading "." or leading "$" in
```
[the remark this answers is not in the document]
Comment (bhtowles) [reply tests/test_mongodb_fname_restrictions.py:31]: Not a bug in the tap but certainly a mongodb limitation said to be addressed in future versions.

File: tests/test_mongodb_fname_restrictions.py (hunk 2 above)
Comment (bhtowles) [reply tests/test_mongodb_fname_restrictions.py:278]: Will add

File: tests/test_mongodb_fname_restrictions.py (hunk 1 above)
Comment (bhtowles) [reply tests/test_mongodb_fname_restrictions.py:248]: Updated to insert and update for both special characters (although the '.' is not supported for updates in this mongo version(documented in TDL20088).

**Review by kspeer825 — APPROVED** [review]
